daemon/server/router/system/info_response_test.go

```diff
@@ -31,3 +34,58 @@ func TestLegacyFields(t *testing.T) {
 		t.Errorf("legacy fields should contain %s: %s", expected, string(data))
 	}
 }
+
+// TestMarshalRegistryConfigLegacyFields verifies extra fields in the registry config
+// field in the info response are serialized if they are not empty.
+// This is used for backwards compatibility for API versions < 1.47.
+func TestMarshalRegistryConfigLegacyFields(t *testing.T) {
+	expected := []string{
+		`"AllowNondistributableArtifactsCIDRs": null`,
+		`"AllowNondistributableArtifactsHostnames": null`,
+	}
+
+	tests := []struct {
+		name   string
+		info   *infoResponse
+		assert func(t *testing.T, data []byte, err error)
+	}{
+		{
+			name: "without legacy fields",
+			info: &infoResponse{
+				Info: &system.Info{},
+			},
+			assert: func(t *testing.T, data []byte, err error) {
+				assert.NilError(t, err)
+				for _, v := range expected {
+					assert.Check(t, !is.Contains(string(data), v)().Success())
```

Comment:
String comparisons? As an alternative, you could [unmarshal into a `map[string]any`](https://go.dev/play/p/hVQdLtLwp27) and assert on the presence/absence of map keys.

Reply:
Yep that looks much better. :)